sdk/core/azure-core-test/src/main/java/com/azure/core/test/annotation/TestingServiceVersions.java

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.test.annotation;
+
+import com.azure.core.test.TestMode;
+import com.azure.core.util.ServiceVersion;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation provides test classes the ability to configure which service versions are used when running tests in
+ * {@link TestMode#RECORD}, {@link TestMode#LIVE}, and {@link TestMode#PLAYBACK} modes.
+ * <p>
+ * The service versions configured here mat interact with how {@link AzureMethodSource} generates
+ * parameterized test permutations.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE})
+public @interface TestingServiceVersions {
```

Comment:
Can this be combined with the previous annotation `AzureMethodSource` instead of having 2 separate annotations that can potentially have mutually exclusive versions?

Reply:
The best option that may be available is for `AzureMethodSource` to have another few properties which match what is in `TestingServiceVersions`. There will still need to be separation as JUnit, as far as I've found, doesn't have the ability to parameterize tests based on a class-level annotation.